### ISSUE TITLE
Bad initial completion ranking (no prefix) for named arguments

### ISSUE THREAD

**Issue by DanTup** [Dart-Code/Dart-Code#4618]
When an argument is positional, this initial ranking is good:

![Screenshot 2023-06-28 175956](https://github.com/Dart-Code/Dart-Code/assets/1078012/ec28cc28-57f3-4e93-b1b3-c2281e40e9b2)

However if the argument is named, this ranking is bad:

![Screenshot 2023-06-28 180053](https://github.com/Dart-Code/Dart-Code/assets/1078012/e1f7372b-76d4-4e2b-ae37-987ebb3fe90b)

**Comment by DanTup** [comment]
Seems like this is happening on the server, the rankings are different between the two cases, and the ArgumentError constructor comes above the enum values in the case where there's an argument name:

![image](https://github.com/Dart-Code/Dart-Code/assets/1078012/86bd0377-be17-4cec-8218-f9cb171495e1)

@bwilkerson before I look into the ranking, can you see any reason why the ranking would differ for the value of a named argument versus a positional one? I feel like these things should be identical, but I might be overlooking something.

**Comment by bwilkerson** [comment]
Sorry, but no, I don't know why the ranking is the way it is.

If it helps, there's a class named `SuggestionListener` that can be used by tests to capture information about a completion suggestion, such as the values of the features that are used to compute the ranking. That might help you identify what's causing these rankings to be what they are.

If I had to guess I'd start by looking at the `elementKind`, which is a result of a look-up in a "relevance table". The tables were built several years ago and are woefully out of date. They were built by analyzing actual code and computing the probability of various kinds of suggestions, and the process is easily skewed by the kind of code being analyzed.

Unfortunately, updating the relevance tables isn't trivial, but just removing their use had (last time I measured) a significant negative impact on the quality of completion results.

**Comment by DanTup** [comment]
You're right - this seems to be a difference in the `elementKind` weightings for "ArgumentList_method_unnamed" and "ArgumentList_method_named":

```
'ArgumentList_method_named': {
  ElementKind.ENUM: ProbabilityRange(lower: 0.033, upper: 0.055),
  ElementKind.CONSTRUCTOR: ProbabilityRange(lower: 0.760, upper: 1.000),
},
'ArgumentList_method_unnamed': {
  ElementKind.ENUM: ProbabilityRange(lower: 0.025, upper: 0.035),
  ElementKind.CONSTRUCTOR: ProbabilityRange(lower: 0.188, upper: 0.279),
},
```

I suspect this isn't a big issue (since most people will start typing _something_, at which point VS Code re-ranks), though I'm curious what's involved in updating the relevance tables (and how risky it might be) :-)

**Comment by DanTup** [comment]
Not sure if it's exactly the same issue (or if this is actually an issue), but recording it here for reference - literal booleans appear much further down the list for boolean named arguments than I would've expected (my feeling is that literal booleans would be fairly common in arguments like this).

![image](https://github.com/Dart-Code/Dart-Code/assets/1078012/0495c115-dab7-4021-ad1d-a1b2072e7bda)
